### ISSUE TITLE
Set HTTP Headers Dynamically?

### ISSUE THREAD

**Issue by MCKLtech** [SocialiteProviders/Providers#323]
There is one case that I've discovered where the Content-Type header must be set. For example, in the Salesforce provider with Lightning services, the HTTP _Content-Type_ must be _application/x-www-form-urlencoded_

If it isn't, the result is a **unsupported_grant_type**

Inspecting the error, the _Accept_ header is set to _application/json_ but having tried the following:

```
$clientId = "secret";
$clientSecret = "secret";
$redirectUrl = "http://yourdomain.com/api/redirect";
$additionalProviderConfig = ['headers' => array('Content-Type' => 'application/x-www-form-urlencoded')];
$config = new \SocialiteProviders\Manager\Config($clientId, $clientSecret, $redirectUrl, $additionalProviderConfig);
return Socialite::with('salesforce')->setConfig($config)->redirect();
```

There is no effect. Digging into the codebase, it would appear _league_ and it's _OAuth1_ method of _getHeaders()_ set the headers and they can't be changed on the fly.

TL;DR - Is it possible to change or add headers dynamically?

**Comment by m1guelpf** [comment]
It should be possible to add some kind of setter to the Manager underlying package. I'd be open to accepting a PR for this :)
